sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -129,6 +130,21 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     }
   }
 
+  /**
+   * For Hive serde tables, some serde libraries require the specified schema and record it
+   * in the metastore. The main code is from the function hasMetastoreBasedSchema in Table.java
```

Comment:
To the reviewers: `hasMetastoreBasedSchema` in `Table.java` can be accessed via the link:  https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/metadata/Table.java#L943-L946